llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -3533,6 +3533,36 @@ AArch64TTIImpl::getCostOfKeepingLiveOverCall(ArrayRef<Type *> Tys) {
   return Cost;
 }
 
+bool AArch64TTIImpl::isPartialReductionSupported(
+    const Instruction *ReductionInstr, Type *InputType, unsigned ScaleFactor,
```

Comment:
What is `ScaleFactor` ?

Reply:
It's the scaling difference between the element size of the inputs and the size of the resulting scalar from the entire reduction. In the tests you'll see that the input types are extended and the resulting scalar is 4 times the size, which mirrors how the dot product instructions work.

Reply:
That's correct, thanks for spotting that. It's not used in this PR but is used by the loop vectorizer part of this work, so I'll move it to that PR.